modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+            List<Assignment> assignments = postAssignments(application, partialpage, new User("user" + i));
+            for (Assignment assignment : assignments) {
+                Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+
+            }
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state and the remaining
+        // should be in running state
+        List<Assignment> assignments = postAssignments(application, partialpage, new User("user1000"));
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            Experiment exp = getExperiment(batchExperiments.get(i - 1));
+            if (i <= NUMBER_OF_RAPID_EXPERIMENTS_IN_BATCH) {
+                Assert.assertEquals(assignments.get(i - 1).status, "EXPERIMENT_PAUSED");
+                Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+            } else {
+                Assert.assertEquals(assignments.get(i - 1).status, "NEW_ASSIGNMENT");
+                Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+            }
+
+        }
+
+    }
+
+    /**
+     * This test case covers a scenario where we do batch assignment for a group of experiments that has all rapid
+     * experiments. It validates the behavior when we do bulk assignment of a user to that batch
+     */
+    @Test
+    public void testAssignmentForBatchWithAllRapidExperiments() {
+
+        page = new Page(BATCH_PAGE, true);
+        List<Experiment> expList = new ArrayList<Experiment>();
+        // lets create experiments for the batch
+        for (int i = 1; i <= NUMBER_OF_EXPERIMENTS_PER_BATCH; i++) {
+            Experiment exp = ExperimentFactory.createExperiment("_" + UUID.randomUUID().toString(), -1)
+                    .setApplication(application).setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+
+            exp = postExperiment(exp);
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 3);
+            postBuckets(bucketList);
+
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            exp = putExperiment(exp);
+            expList.add(exp);
+
+        }
+
+        assignPageToExperimentsList(expList, page);
+
+        // the assignment will happen until the max users cap is reached for all experiments in batch
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            List<Assignment> assignments = postAssignments(application, page, new User("user" + (i + 1000)));
+            for (Assignment assignment : assignments) {
+                Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+
+            }
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state and the remaining
+        // should be in running state
+        List<Assignment> assignments = postAssignments(application, page, new User("user100"));
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
```

Comment:
@sudheerkethireddy I don't think that is the best idea. You should check for the experiment state and the assignment status separately as they are disjoint with each other. The `assignments` list and the `expList` are 2 different list data structures and may vary in their sizes and position of elements. As long as you have the `RAPID_EXP_MAX_USERS` same as `NUMBER_OF_EXPERIMENTS_PER_BATCH`, this will continue to work but as soon as these 2 values get changed in future and diverge, you will end up hitting `ArrayIndexOutOfBoundsException`. 

For example, say `RAPID_EXP_MAX_USERS` gets modified to 20 and the number of experiments stays at 5 then you will end up trying to retrieve a value at index 6-19 on expList which will throw an `ArrayIndexOutOfBoundsException`.

We can sync up offline if you don't understand.